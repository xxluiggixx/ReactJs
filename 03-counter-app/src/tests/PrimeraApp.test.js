const { render, getByText } = require("@testing-library/react");
const { default: PrimeraApp } = require("../PrimeraApp");
import React from "react";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";

describe("Pruebas en <PrimeraApp/>", () => {
  /*   test('should show message "Hola, Soy Goku" ', () => {
    const saludo = "Hola, soy Goku";
    const { getByText } = render(<PrimeraApp saludo={saludo} />);

    expect(getByText(saludo)).toBeInTheDocument();
  }); */
  test("Debe de mostrar <PrimeraApp/> correcatamente", () => {
    const saludo = "Hola, soy Goku!";
    const wrapper = shallow(<PrimeraApp saludo={saludo} />);
    expect(wrapper).toMatchSnapshot();
  });
  test("Debe de mostrar el subtitulo enviado por props", () => {
    const subtitulo = "Soy un subtitulo";
    const saludo = "Hola, soy Goku!";
    const wrapper = shallow(
      <PrimeraApp saludo={saludo} subtitulo={subtitulo} />
    );
    const textoParrafo = wrapper.find("p").text();
    expect(textoParrafo).toBe(subtitulo);
  });
});
