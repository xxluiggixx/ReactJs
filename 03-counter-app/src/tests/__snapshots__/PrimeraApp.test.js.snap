// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimeraApp/> Debe de mostrar <PrimeraApp/> correcatamente 1`] = `
<Fragment>
  <h1>
    Hola, soy Goku!
  </h1>
  <p>
    soy un subtitulo
  </p>
</Fragment>
`;
